lib/manageiq/api/client/client.rb

```diff
@@ -31,6 +30,25 @@ def load_definitions
         @collections   = load_collections(entrypoint["collections"])
       end
 
+      def update_authentication(auth_options = {})
+        return @authentication unless ManageIQ::API::Client::Authentication.options?(auth_options)
+        saved_auth = @authentication
+        @authentication = ManageIQ::API::Client::Authentication.new(auth_options)
+        begin
+          reconnect
+        rescue => err
+          @authentication = saved_auth
+          raise err
```

Comment:
Prefer `raise` over `raise err`.  The former will not impact the stacktrace while the latter will.